halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/CustomSizing.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017 Target, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.config.model.v1.node;
+
+import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemSetBuilder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A CustomSizing is a map of maps where you can hack in provider-specific settings related to instance/container/pod sizes.
+ */
+@Data
+@EqualsAndHashCode(callSuper = false)
+public class CustomSizing implements Map<String, Map> {
```

Comment:
Is the intent here just to provide a type alias?

Reply:
Heh. Not really.

Initially this class had some structure to it, because it assumed the kubernetes "requests" and "limits" nomenclature. We also extended Node initially, and then removed that after we realized we had nothing to validate because there's no CLI interaction with this data (and we couldn't think of any reasonable validation that wasn't k8s-specific). By the time we removed those things, this had nothing left but the Map implementation; you were the first to realize this. :-)

But now, we've given this class a new reason to exist: it holds our goofy little ```stringOrNull``` method, so that it doesn't have to be duplicated in the manual and orca deployment flows.

(We abandoned the idea of explicitly making this a three-layer map with strings at its leaf nodes. No implementation felt clean, and we also realized that when we add replicaSetSize to it later this week, that will be an integer, at the top tier of the map. So we'd be breaking that constraint ourselves.)

Reply:
Hmm it might still be valuable to validate even if there is no CLI command to generate entries here - but maybe not. 

Sounds good

Reply:
We'd be wide open to it if you can guide us on _what_ to validate in a way that doesn't assume a specific cloud provider, or on a way to detect that we're deploying to k8s and only then do our k8s-specific validations. I would definitely feel better if we validated some basics in the k8s case.

On a later PR, of course. :-)